docs/INTEGRATION_TUTORIAL.md

```diff
@@ -508,6 +866,131 @@ jobs:
           github-token: ${{ secrets.GITHUB_TOKEN }}
 ```
 
+## What Success Looks Like: End-to-End Flow
+
+Here's what a complete, successful git-perf integration looks like in action:
+
+### Step-by-Step Walkthrough
+
+1. **You make a code change** that affects performance (e.g., optimize a function)
+
+2. **Create a pull request**:
+   ```bash
+   git checkout -b optimize-parser
+   git commit -am "perf: optimize JSON parser"
+   git push -u origin optimize-parser
+   gh pr create --title "perf: optimize JSON parser"
+   ```
+
+3. **GitHub Actions automatically runs**:
+   - ✅ Checks out code with full history (`fetch-depth: 0`)
+   - ✅ Installs git-perf
+   - ✅ Configures git identity
+   - ✅ Builds project and measures `build_time`
+   - ✅ Runs tests and measures `test_duration`
+   - ✅ Measures `binary_size`
+   - ✅ Pushes measurements to `refs/notes/perf-v3`
+   - ✅ Generates interactive HTML report
+   - ✅ Publishes report to GitHub Pages
+   - ✅ Runs audit to check for regressions
+   - ✅ Comments on your PR with results
+
+4. **You receive a PR comment** with performance analysis:
+
+   ```markdown
+   ## Performance Report
+
+   ⏱  [Performance Results](https://username.github.io/repo/abc123def.html)
+
+   ## Audit Results
+
+   ✅ 'build_time' (seconds)
+   z-score (mad): ↓ 2.15
+   Head: μ: 38.2 σ: 0.5 MAD: 0.3 n: 1
+   Tail: μ: 42.1 σ: 1.8 MAD: 1.2 n: 15
+    [-9.3% improvement] ▃▅▄▆▅▄▅▃▅▂↓
+
+   ✅ 'test_duration' (ms)
+   z-score (mad): → 0.45
+   Head: μ: 1250 σ: 12 MAD: 8 n: 1
+   Tail: μ: 1248 σ: 25 MAD: 18 n: 15
+    [+0.2% – within normal variance] ▃▅▄▆▅▄▅▃▅▄
+
+   ❌ 'binary_size' (bytes)
+   z-score (mad): ↑ 5.23
+   Head: μ: 4,823,552 σ: 0 MAD: 0 n: 1
+   Tail: μ: 4,512,128 σ: 8,192 MAD: 4,096 n: 15
+    [+6.9% – REGRESSION DETECTED] ▃▅▄▆▅▄▅▃▅↑
+
+   _Created by [git-perf](https://github.com/kaihowl/git-perf/)_
+   ```
+
+5. **You analyze the results**:
+   - ✅ Build time improved by 9.3% - Great!
+   - ✅ Test duration unchanged - Expected
+   - ❌ Binary size increased by 6.9% - Needs investigation
+
+6. **You click the report link** to see the interactive dashboard:
+   - View historical trends with Plotly charts
+   - Filter by branch, measurement, or time range
+   - See sparklines showing performance over time
+   - Export data as CSV for further analysis
+
+7. **You investigate the binary size regression**:
+   ```bash
+   # Check what changed
+   git diff main...optimize-parser -- Cargo.lock
+
+   # Turns out the optimization added a new dependency
+   # You decide the performance gain is worth the size increase
+   # Or you find a way to achieve the optimization without the dependency
```

Comment:
The step to accept the performance regression is to bump the epoch for the specific measurement